src/python/grpcio_tests/tests_aio/unit/interceptor_test.py

```diff
@@ -0,0 +1,556 @@
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+
+            await server.stop(None)
+
+            call = multicallable(messages_pb2.SimpleRequest(), timeout=0.1)
+
+            with self.assertRaises(aio.AioRpcError) as exception_context:
+                await call
+
+            self.assertTrue(call.done())
+            self.assertFalse(call.cancelled())
+            self.assertEqual(await call.code(),
+                             grpc.StatusCode.DEADLINE_EXCEEDED)
+            self.assertEqual(await call.details(), 'Deadline Exceeded')
+            self.assertEqual(await call.initial_metadata(), ())
+            self.assertEqual(await call.trailing_metadata(), ())
+
+    async def test_call_rpcerror_awaited(self):
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                call = await continuation(client_call_details, request)
+                await call
+                return call
+
+        server_target, server = await start_test_server()  # pylint: disable=unused-variable
+
+        async with aio.insecure_channel(server_target,
+                                        interceptors=[Interceptor()
+                                                     ]) as channel:
+
+            multicallable = channel.unary_unary(
+                '/grpc.testing.TestService/UnaryCall',
+                request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                response_deserializer=messages_pb2.SimpleResponse.FromString)
+
+            await server.stop(None)
+
+            call = multicallable(messages_pb2.SimpleRequest(), timeout=0.1)
+
+            with self.assertRaises(aio.AioRpcError) as exception_context:
+                await call
+
+            self.assertTrue(call.done())
+            self.assertFalse(call.cancelled())
+            self.assertEqual(await call.code(),
+                             grpc.StatusCode.DEADLINE_EXCEEDED)
+            self.assertEqual(await call.details(), 'Deadline Exceeded')
+            self.assertEqual(await call.initial_metadata(), ())
+            self.assertEqual(await call.trailing_metadata(), ())
+
+    async def test_cancel_before_rpc(self):
+
+        interceptor_reached = asyncio.Event()
+
+        class Interceptor(aio.UnaryUnaryClientInterceptor):
+
+            async def intercept_unary_unary(self, continuation,
+                                            client_call_details, request):
+                interceptor_reached.set()
+                await asyncio.sleep(0)
```

Comment:
Can we use a never resolving object (e.g. a future, or a event, or infinite sleep) to await here? By reading the code, the intention of this test is cancelling the interception while it is handling another async job. It could protect us against future change in the AsyncIO implementation.